btf/btf_types.go

```diff
@@ -300,6 +326,17 @@ const (
 	btfIntBitsShift     = 0
 )
 
+var btfIntLen = packedSize[btfInt]()
+
+func unmarshalBtfInt(bi *btfInt, b []byte, bo binary.ByteOrder) (int, error) {
```

Comment:
Why not make these methods on `btfInt` and so on?

Reply:
That works for the single types, but for the types where we have slices such as `[]btfMember` and `[]btfParam` its nice to have a function to marshal the whole slice and do the bounds checks ect.

I thought this worked just as well.

We can also go the method route but we would have to declare a type alias for the slices to be able to add methods to them.

```
type btfMembers []btfMember

func (bm btfMembers) Marshal(b []byte, bo binary.ByteOrder) (int, error) { ....
```

But that seemed like to much added code for the same effect.

Reply:
Fair enough. Medium term we need to get rid of all of these manual marshalers anyways.